sdk/cosmosdb/cosmos/test/public/functional/continuation-token-complete.spec.ts

```diff
@@ -0,0 +1,2733 @@
+    expectedTokenStructure: {
+      expectNoContinuationToken: true,
+    },
+    expectedTokenValues: {
+      expectUndefined: true,
+    },
+    tokenParser: (_token) => null, // No token expected
+    validator: (_parsed) => true, // No validation needed for undefined tokens
+    requiresMultiPartition: true,
+    description:
+      "GROUP BY queries don't support continuation tokens as they require full aggregation",
+  },
+
+  // ============= COMPLEX QUERIES =============
+  {
+    name: "JOIN with ORDER BY",
+    query: "SELECT c.id, c.name, t FROM c JOIN t IN c.tags ORDER BY c.id",
+    queryOptions: { maxItemCount: 2, enableQueryControl: true },
+    expectedTokenStructure: {
+      hasCompositeToken: true,
+      hasOrderByItems: true,
+      hasSkipCount: true,
+      hasRid: true,
+    },
+    expectedTokenValues: {
+      ridType: "string",
+      orderByItemsCount: 1,
+      skipCountInitial: 2,
+    },
+    tokenParser: (token) => JSON.parse(token),
+    validator: (parsed) => {
+      return (
+        parsed.rangeMappings &&
+        Array.isArray(parsed.rangeMappings) &&
+        parsed.orderByItems &&
+        Array.isArray(parsed.orderByItems) &&
+        parsed.orderByItems.length > 0
+      );
+    },
+    requiresMultiPartition: true,
+    description: "JOIN with ORDER BY should produce OrderBy continuation tokens",
+  },
+
+  {
+    name: "WHERE with ORDER BY",
+    query: "SELECT * FROM c WHERE c.amount > 20 ORDER BY c.amount ASC",
+    queryOptions: { maxItemCount: 3, enableQueryControl: true },
+    expectedTokenStructure: {
+      hasRangeMappings: true,
+      hasOrderByItems: true,
+      hasSkipCount: true,
+      hasRid: true,
+    },
+    expectedTokenValues: {
+      ridType: "string",
+      orderByItemsCount: 1,
+      skipCountInitial: 1,
+    },
+    tokenParser: (token) => JSON.parse(token),
+    validator: (parsed) => {
+      return parsed.rangeMappings && parsed.orderByItems && typeof parsed.skipCount === "number";
+    },
+    requiresMultiPartition: true,
+    description: "Filtered ORDER BY queries should maintain ordering with predicates",
+  },
+];
+
+// TODO: remove console.log from tests
+describe("Comprehensive Continuation Token Tests", { timeout: 120000 }, () => {
+  let singlePartitionContainer: Container;
+  let multiPartitionContainer: Container;
+  let multiPartitionContainer2: Container;
+
+  beforeAll(async () => {
+    await removeAllDatabases(client);
+
+    // Create single partition container with essential composite indexes
+    singlePartitionContainer = await getTestContainer(
+      "single-partition-test",
+      client,
+      {
+        partitionKey: { paths: ["/pk"] },
+        throughput: 1000,
+        indexingPolicy: {
+          indexingMode: "consistent",
+          automatic: true,
+          includedPaths: [{ path: "/*" }],
+          excludedPaths: [{ path: '/"_etag"/?' }],
+          compositeIndexes: [
```

Comment:
nit: extract composite index in a variable and reuse

Reply:
it's not reused